pulser-core/pulser/waveforms.py

```diff
@@ -805,12 +805,12 @@ def duration(self) -> int:
     @cached_property
     def _samples(self) -> np.ndarray:
         """The value at each time step that describes the waveform."""
-        return cast(
-            np.ndarray,
-            np.round(
-                self._interp_func(np.arange(self._duration)), decimals=9
-            ),  # Rounds to the order of Hz
-        )
+        samples = self._interp_func(np.arange(self._duration))
+        value_range = np.max(np.abs(samples))
```

Comment:
nit: `np.abs(samples).max()` seems to be more concise and faster...
![image](https://github.com/pasqal-io/Pulser/assets/61453516/1de7cae1-1a30-4683-bed4-40917a2ae729)
![image](https://github.com/pasqal-io/Pulser/assets/61453516/333e1e6d-cbd4-44b8-8046-3631fbd3b37f)
![image](https://github.com/pasqal-io/Pulser/assets/61453516/e3d0980c-3ed1-4c74-93b9-ef1d60088886)